### PR TITLE
remove base amounts usage

### PR DESCRIPTION
**Pull request by belbix**

### CODE REVIEW

File: contracts/strategies/ConverterStrategyBase.sol [hunk 1]
```diff
@@ -339,12 +324,6 @@ abstract contract ConverterStrategyBase is ITetuConverterCallback, DepositorBase
     uint[] memory withdrawnAmounts = _depositorEmergencyExit();
     emit OnDepositorEmergencyExit(withdrawnAmounts);
 
-    (address[] memory tokens, uint indexAsset) = _getTokens(asset);
-
-    // convert amounts to main asset and update base amounts
-    (uint collateral, uint[] memory repaid) = _convertAfterWithdrawAll(tokens, indexAsset);
```
Comment (dvpublic) [inline contracts/strategies/ConverterStrategyBase.sol:345]: there is same problem here. After emergency exist we should have only underlying on the balance, conversion is necessary

File: contracts/strategies/ConverterStrategyBase.sol [hunk 2]
```diff
@@ -638,7 +571,7 @@ abstract contract ConverterStrategyBase is ITetuConverterCallback, DepositorBase
     v.len = v.tokens.length;
 
     // get amount of target asset available to be sent
-    v.theAssetBaseAmount = baseAmounts[theAsset_];
+    v.theAssetBaseAmount = _balance(theAsset_);
```
Comment (dvpublic) [inline contracts/strategies/ConverterStrategyBase.sol:579]: I would suggest to rename base amounts to balances in such cases

File: contracts/strategies/ConverterStrategyBaseLib.sol [hunk 3]
```diff
@@ -278,7 +272,7 @@ library ConverterStrategyBaseLib {
     for (uint i; i < len; i = AppLib.uncheckedInc(i)) {
       if (i == indexAsset) continue;
 
-      uint baseAmount = baseAmounts_[tokens[i]];
+      uint baseAmount = IERC20(tokens[i]).balanceOf(address(this));
```
Comment (dvpublic) [inline contracts/strategies/ConverterStrategyBaseLib.sol:275]: probably we should rename baseAmount to balance

File: contracts/strategies/uniswap/UniswapV3ConverterStrategy.sol [hunk 4]
```diff
@@ -240,12 +222,11 @@ contract UniswapV3ConverterStrategy is UniswapV3Depositor, ConverterStrategyBase
         uint[] memory tokenAmounts = new uint[](2);
         tokenAmounts[0] = amount_;
         emit OnDepositorEnter(tokenAmounts, tokenAmounts);
-        _updateBaseAmountsForAsset(state.tokenA, amount_, 0);
       } else {
         (address[] memory tokens, uint indexAsset) = _getTokens(asset);
 
         // prepare array of amounts ready to deposit, borrow missed amounts
-        (uint[] memory amounts, uint[] memory borrowedAmounts, uint collateral) = _beforeDeposit(
+        (uint[] memory amounts,,) = _beforeDeposit(
```
Comment (dvpublic) [inline contracts/strategies/uniswap/UniswapV3ConverterStrategy.sol:235]: borrowedAmounts and collateral are not used anymore, probably _beforeDeposit (and getTokenAmounts) shouldn't return that values to reduce gas consumption.

File: contracts/strategies/ConverterStrategyBase.sol [hunk 5]
```diff
@@ -322,10 +311,6 @@ abstract contract ConverterStrategyBase is ITetuConverterCallback, DepositorBase
         );
       }
 
-      // convert amounts to main asset and update base amounts
-      (uint collateral, uint[] memory repaid) = _convertAfterWithdraw(tokens, indexAsset, vars.amountsToConvert, _converter);
```
[the remark this answers is not in the document]
Comment (belbix) [reply contracts/strategies/ConverterStrategyBase.sol:326]: mistake, fixed

**Pull request merged**